internal/provider/resource_auth0_resource_server.go

```diff
@@ -203,40 +196,70 @@ func updateResourceServer(ctx context.Context, d *schema.ResourceData, m interfa
 
 func deleteResourceServer(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	api := m.(*management.Management)
+
 	if err := api.ResourceServer.Delete(d.Id()); err != nil {
-		if mErr, ok := err.(management.Error); ok {
-			if mErr.Status() == http.StatusNotFound {
-				d.SetId("")
-				return nil
-			}
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			d.SetId("")
+			return nil
 		}
+		return diag.FromErr(err)
 	}
 
+	d.SetId("")
 	return nil
 }
 
 func expandResourceServer(d *schema.ResourceData) *management.ResourceServer {
+	config := d.GetRawConfig()
+
 	resourceServer := &management.ResourceServer{
-		Name:                 String(d, "name"),
-		Identifier:           String(d, "identifier"),
-		SigningAlgorithm:     String(d, "signing_alg"),
-		SigningSecret:        String(d, "signing_secret", IsNewResource(), HasChange()),
-		AllowOfflineAccess:   Bool(d, "allow_offline_access"),
-		TokenLifetime:        Int(d, "token_lifetime"),
-		TokenLifetimeForWeb:  Int(d, "token_lifetime_for_web"),
-		VerificationLocation: String(d, "verification_location"),
-		Options:              Map(d, "options"),
-		EnforcePolicies:      Bool(d, "enforce_policies"),
-		TokenDialect:         String(d, "token_dialect", IsNewResource(), HasChange()),
```

Comment:
For signing secret and token dialect we had both `IsNewResource` and `HasChange` checks which didn't really make sense because that means those 2 values can always be changed and updated so we don't really need the conditionals.